ext/standard/dns.c

```diff
@@ -169,34 +169,52 @@ PHP_FUNCTION(gethostbyaddr)
 static zend_string *php_gethostbyaddr(char *ip)
 {
 #if HAVE_IPV6 && HAVE_INET_PTON
-	struct in6_addr addr6;
-#endif
-	struct in_addr addr;
-	struct hostent *hp;
+	struct sockaddr_in sa4;
+	struct sockaddr_in6 sa6;
+	char out[NI_MAXHOST], *p;
 
-#if HAVE_IPV6 && HAVE_INET_PTON
-	if (inet_pton(AF_INET6, ip, &addr6)) {
-		hp = gethostbyaddr((char *) &addr6, sizeof(addr6), AF_INET6);
-	} else if (inet_pton(AF_INET, ip, &addr)) {
-		hp = gethostbyaddr((char *) &addr, sizeof(addr), AF_INET);
-	} else {
-		return NULL;
+	if (inet_pton(AF_INET6, ip, &sa6.sin6_addr)) {
+		sa6.sin6_family = AF_INET6;
+
+		if (getnameinfo((struct sockaddr *)&sa6, sizeof(sa6), out, sizeof(out), NULL, 0, 0) < 0) {
+			return zend_string_init(ip, strlen(ip), 0);
+		}
+		p = strchr(out, '%'); /* demangle numeric host with %name suffix */
```

Comment:
Any context on that? Hard to search for.

Reply:
You have the same piece of code in 
https://github.com/php/php-src/blob/master/ext/standard/net.c#L85

Also observed in test, without this check 
```
$ sapi/cli/php -r 'var_dump(gethostbyaddr("::2"));'
string(14) "::2%4294967169"

```

Reply:
seems this only happen for IPv6 when not found
Using NI_NAMEREQD seems cleaner